src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -259,11 +260,11 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
     // For now we need to update our internal `lastL1BlockNumber` value
     // which is used when submitting batches.
     this._updateLastL1BlockNumber() // TODO: Remove this
-    let batch: Batch = []
-    for (let i = startBlock; i < endBlock; i++) {
-      this.log.debug(`Fetching L2BatchElement ${i}`)
-      batch.push(await this._getL2BatchElement(i))
-    }
+    const blockRange = endBlock - startBlock
+    let batch: Batch = await bPromise.map([...Array(blockRange).keys()], (i) => {
+      return this._getL2BatchElement(startBlock + i)
+    }, {concurrency: 50})
+
```

Comment:
this preserves the order from just logging in the tests, and makes our test like 20% faster which is p nice!